### ISSUE TITLE
"pod install" causes noise in SPM Package.resolved file from Xcode project

### ISSUE THREAD

**Issue by dnedrow** [CocoaPods/CocoaPods#12014]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
1. Create a new Xcode iOS App project called "Foo" with "Git Repository" selected.
3. Add `https://github.com/Quick/Nimble` as a Swift Package dependency to the project.
4. Wait for packages to resolve.
5. Commit the project
6. Create a Podfile and add any dependency, e.g., `MaterialComponents/TextControls+FilledTextFields`
7. Run `pod install`.
8. Examine the diff. You should see that the pod installation actually changed the case of the existing Swift Package dependency.

## What did you expect to happen?

I expected the pod to be installed without manipulating the Package.resolved file under the `Foo.xcworkspace/xcshareddata/swiftpm` directory

## What happened instead?

Inexplicably, `pod install` manipulated the contents of the Package.resolved file.

### Stack

```
   CocoaPods : 1.12.1
        Ruby : ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [arm64-darwin22]
    RubyGems : 3.2.3
        Host : macOS 13.4.1 ((c))
       Xcode : 14.3.1 (14E300c)
         Git : git version 2.41.0
Ruby lib dir : /Users/n1587127/.rvm/rubies/ruby-3.0.0/lib
Repositories : ApolloSPM - git - https://github.com/dnedrow/apollo-ios @ 3bd8cf82279ce9b5da3506850ab5e4487db946e9

               salemove-glia-ios-podspecs - git - https://github.com/salemove/glia-ios-podspecs.git @ 47180c566517919b4eb1524e676587b17b60d53f

               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /Users/n1587127/.rvm/gems/ruby-3.0.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.5
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
```

**Comment by iosdevben** [comment]
Did you create the Xcode app project 'Foo' with the same version of Xcode as that which was used when `pod install` is run?

**Comment by dnedrow** [comment]
Yes, same version of Xcode for both. Does `pod install` trigger a SPM resolution?

**Comment by DanilChenikov** [comment]
Has someone found any solution? 
Just because every time after 'pod install' command, my SPM' Package.resolved is broken

**Comment by tichise** [comment]
"I am currently facing a similar issue."

**Comment by dclcs** [comment]
I have same problem

**Comment by MahdiRB** [comment]
Same problem

**Comment by SMA-HP** [comment]
same here!

**Comment by app-harry** [comment]
same here!
